integration-test/src/test/java/org/apache/iotdb/confignode/it/IoTDBClusterRestartIT.java

```diff
@@ -83,5 +138,137 @@ public void clusterRestartTest() throws InterruptedException {
     ((AbstractEnv) EnvFactory.getEnv()).testWorking();
   }
 
+  @Test
+  public void clusterRestartAfterUpdateDataNodeTest() throws InterruptedException {
+    TShowClusterResp clusterNodes;
+    final String sg = "root.sg";
+    final String sg0 = "root.sg0";
+    final String sg1 = "root.sg1";
+
+    final String d00 = sg0 + ".d0.s";
+    final String d01 = sg0 + ".d1.s";
+    final String d10 = sg1 + ".d0.s";
+    final String d11 = sg1 + ".d1.s";
+
+    final String allPaths = "root.**";
+    final String allSg0 = "root.sg0.**";
+    final String allSg1 = "root.sg1.**";
+    // Shutdown all data nodes
+    for (int i = 0; i < testDataNodeNum; i++) {
+      EnvFactory.getEnv().shutdownDataNode(i);
+    }
+
+    // Sleep 1s before restart
+    TimeUnit.SECONDS.sleep(1);
+
+    // Modify data node config
+    List<DataNodeWrapper> dataNodeWrapperList = EnvFactory.getEnv().getDataNodeWrapperList();
+    List<ConfigNodeWrapper> configNodeWrappersList = EnvFactory.getEnv().getConfigNodeWrapperList();
+    for (int i = 0; i < testDataNodeNum; i++) {
+      int[] portList = EnvUtils.searchAvailablePorts();
+      dataNodeWrapperList.get(i).setPort(portList[0]);
+      dataNodeWrapperList.get(i).setInternalPort(portList[1]);
+      dataNodeWrapperList.get(i).setMppDataExchangePort(portList[2]);
+
+      // update data node files'names
+      dataNodeWrapperList.get(i).renameFile();
+    }
+
+    for (int i = 0; i < testDataNodeNum; i++) {
+      dataNodeWrapperList.get(i).changeConfig(ConfigFactory.getConfig().getEngineProperties());
+      EnvFactory.getEnv().startDataNode(i);
+    }
+
+    ((AbstractEnv) EnvFactory.getEnv()).testWorking();
+
+    // check nodeInfo in cluster
+    try (SyncConfigNodeIServiceClient client =
+        (SyncConfigNodeIServiceClient) EnvFactory.getEnv().getLeaderConfigNodeConnection()) {
+      // check the number and status of nodes
+      clusterNodes = getClusterNodeInfos(client, testConfigNodeNum, testDataNodeNum);
+      assertEquals(TSStatusCode.SUCCESS_STATUS.getStatusCode(), clusterNodes.getStatus().getCode());
+
+      // check the configuration of nodes
+      List<TConfigNodeLocation> configNodeLocationList = clusterNodes.getConfigNodeList();
+      List<TDataNodeLocation> dataNodeLocationList = clusterNodes.getDataNodeList();
+      checkNodeConfig(
+          configNodeLocationList,
+          dataNodeLocationList,
+          configNodeWrappersList,
+          dataNodeWrapperList);
```

Comment:
We should also check the ip:port that shown by `show cluster`

Reply:
`getClusterNodeInfos()` calls `show cluster`, and `checkNodeConfig` will check the configurations